client/my-sites/themes/helpers.js

```diff
@@ -171,7 +171,9 @@ export function interlaceThemes( wpComThemes, wpOrgThemes, searchTerm, isLastPag
 
 export function getTierRouteParam() {
 	return isEnabled( 'themes/tiers' )
-		? `:tier(${ Object.keys( THEME_TIERS ).join( '|' ) })?`
+		? // We need to come up with some way to handle dynamic tiers. We could either change the route to tier/:tier or
+		  // fetch the tiers from the server and use them to build the route.
+		  `:tier(${ Object.keys( THEME_TIERS ).join( '|' ) })?`
```

Comment:
Nit: Having the comment inside the ternary makes it a little bit difficult to read. In this case I think I'd go for an early `return`:

```suggestion
// We need to come up with some way to handle dynamic tiers. We could either change the route to tier/:tier or
// fetch the tiers from the server and use them to build the route.
if ( isEnabled( 'themes/tiers' ) ) {
	return `:tier(${ Object.keys( THEME_TIERS ).join( '|' ) })?`
}
return ...
```